File: app/src/androidTest/java/com/vutrankien/t9vietnamese/MainActivityTest.java
package com.vutrankien.t9vietnamese;

import android.app.Activity;
import android.content.Context;
import android.support.test.rule.ActivityTestRule;
import android.support.test.runner.AndroidJUnit4;

import com.snappydb.DB;
import com.snappydb.DBFactory;
import com.snappydb.SnappydbException;

import org.junit.Rule;
import org.junit.Test;
import org.junit.runner.RunWith;

import java.util.Arrays;
import java.util.HashSet;

import static org.junit.Assert.*;

@RunWith(AndroidJUnit4.class)
public class MainActivityTest {

    @Rule
    public ActivityTestRule<MainActivity> mActivityRule =
            new ActivityTestRule<>(MainActivity.class);
    @Test
    public void whenFirstStartThenCreateDatabase() throws SnappydbException {
        Activity activity = mActivityRule.getActivity();
        DB snappydb = DBFactory.open(activity, "vi-VN");
        HashSet<String> candidates =  snappydb.get("24236" );//new HashSet<String>().getClass() get array of string

        //assertTrue(new HashSet<String>(candidates), "Jack Reacher");
        assertTrue((candidates).contains("chào"));
        snappydb.close();
    }

}

File: app/src/androidTest/java/com/vutrankien/t9vietnamese/MainActivityTest.kt
package com.vutrankien.t9vietnamese

import androidx.test.espresso.Espresso.onView
import androidx.test.espresso.action.ViewActions.click
import androidx.test.espresso.assertion.ViewAssertions.matches
import androidx.test.espresso.matcher.ViewMatchers.isDisplayed
import androidx.test.espresso.matcher.ViewMatchers.withText
import androidx.test.rule.ActivityTestRule
import androidx.test.runner.AndroidJUnit4
import org.hamcrest.CoreMatchers.*
import org.junit.Rule
import org.junit.Test
import org.junit.runner.RunWith

/**
 * This suppose to contains UI related tests
 */
@RunWith(AndroidJUnit4::class)
class MainActivityTest {
    @get:Rule val mActivityRule = ActivityTestRule<MainActivity>(MainActivity::class.java)

    @Test fun initialize() {
        val engineLoadStr = mActivityRule.activity.getString(R.string.engine_loading)
        val initializedStr = mActivityRule.activity.getString(R.string.notify_initialized)
        onView(anyOf(withText(containsString(engineLoadStr)), withText(containsString("not initialized"))))
                .check(matches(isDisplayed()))
        onView((withText(containsString(initializedStr))))
                .check(matches(isDisplayed()))
    }

    @Test fun basicTyping() {
        "24236".forEach { onView(withText(startsWith("$it"))).perform(click()) }
    }
}